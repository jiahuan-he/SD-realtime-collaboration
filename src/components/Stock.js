import React from 'react'

const stock = {
    stroke: 'black',
    cursor: 'move',
    strokeWidth: "1px",
    fill: 'white'
}

const stockHighlight = {
    stroke: 'green',
    cursor: 'move',
    strokeWidth: "2px",
    fill: 'white'
}

const moveable = {
    cursor: 'move',
}

export default ({posX, posY, id, initValue, highlight}) => 
(
    <g x={posX} y={posY} id={id} style={moveable}>
        <rect x={posX} y={posY} width={"5%"} height={"5%"} style={highlight?stockHighlight:stock}/>
        <foreignObject x={posX} y={posY} width="50" height="50">
            <div>{id+": "+initValue}</div>                 
        </foreignObject>    
    </g>
);