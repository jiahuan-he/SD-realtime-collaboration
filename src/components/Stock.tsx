import React, { CSSProperties } from 'react'

const stock: CSSProperties = {
    stroke: 'black',
    cursor: 'move',
    strokeWidth: "1px",
    fill: 'white'
}

const stockHighlight: CSSProperties = {
    stroke: 'green',
    cursor: 'move',
    strokeWidth: "2px",
    fill: 'white'
}

const moveable: CSSProperties = {
    cursor: 'move',
}

export const Stock: React.FC<{
    posX: number, 
    posY: number, 
    id: string, 
    initValue: number, 
    highlight: boolean}> = ({posX, posY, id, initValue, highlight}) => 
(
    <g x={posX} y={posY} id={id} style={moveable}>
        <rect x={posX} y={posY} width={"5%"} height={"5%"} style={highlight?stockHighlight:stock}/>
        <foreignObject x={posX} y={posY} width="50" height="50">
            <div>{id+": "+initValue}</div>                 
        </foreignObject>    
    </g>
);