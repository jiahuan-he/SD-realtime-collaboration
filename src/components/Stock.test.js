import React from 'react';
import { mount } from 'enzyme'
import Stock from './Stock'
// Avoid Warning: render(): Rendering components directly into document.body is discouraged.
beforeAll(() => {
    const div = document.createElement('div');
    window.domNode = div;
    document.body.appendChild(div);
})

describe('Stock', () => {
    it('should render correctly in "debug" mode', () => {
        const props = {
            "stock": {
                id: "stock0",
                initValue: 0,
                posX: 50,
                posY: 50,
            }
        }
        const stock = mount(<svg>
            <Stock
                stock={props.stock}
                debug
            ></Stock></svg>, { attachTo: window.domNode });
        expect(stock).toMatchSnapshot();
    });
});