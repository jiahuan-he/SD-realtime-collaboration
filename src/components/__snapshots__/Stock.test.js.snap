// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stock should render correctly in "debug" mode 1`] = `
<svg>
  <Stock
    debug={true}
    stock={
      Object {
        "id": "stock0",
        "initValue": 0,
        "posX": 50,
        "posY": 50,
      }
    }
  >
    <g
      id="stock0"
      style={
        Object {
          "cursor": "move",
        }
      }
      x={50}
      y={50}
    >
      <rect
        height="5%"
        style={
          Object {
            "cursor": "move",
            "fill": "white",
            "stroke": "black",
            "strokeWidth": "1px",
          }
        }
        width="5%"
        x={50}
        y={50}
      />
      <foreignObject
        height="50"
        width="50"
        x={50}
        y={50}
      >
        <div>
          stock0: 0
        </div>
      </foreignObject>
    </g>
  </Stock>
</svg>
`;
